lib/src/clojure_lsp/parser.clj

```diff
@@ -27,54 +27,65 @@
        (mapv (fn [loc] (z/find loc z/up edit/top?)))
        (distinct)))
 
+(def ^:private zero-width-space
+  "A unicode character that is incredibly unlikely to be used in regular code.
+  During parsing, used as a valid and easily identified subsitute for what would
+  otherwise be an invalid character."
+  "\u200b")
```

Comment:
not sure a unicode code is the best choice here, not sure this could bring other issues, but it may worth the try

Reply:
Yeah, I hear you. I'd be OK using another character if you have a suggestion. I picked this one because:
1. Code rarely uses any kind of whitespace besides tabs, spaces and linebreaks.
2. I can't think of a reason code would use it anyway, since text editors are supposed to give this character "zero-width", i.e. make it invisible. In typesetting there's probably some use for it, but not in code.
3. rewrite-clj parses it as a symbol, or part of a symbol, not as whitespace.

The fact that rewrite-clj is doing what we want suggests that it (and Java) assume that the code uses unicode characters, which is probably necessary because people put emojis and other stuff in code. I've never looked into how Java tracks how files and strings are encoded—if it works, best not to touch it I guess.